### ISSUE TITLE
Rocketchatctl nodejs version breaks 3.x.x update

### ISSUE THREAD

**Issue by muhv7** [RocketChat/Rocket.Chat#16611]
RocketchatCTL update will break the whole install when trying to update to 3.x.x since the node version being used is way older than 12.x that 3.x RC uses now.

**Comment by grzebo** [comment]
This also breaks the Digital Ocean droplet, which is described in the docs as a supported method of installing.

**Comment by strasharo** [comment]
Confirmed on Ubuntu:
Meteor requires Node v12.0.0 or later.

**Comment by iovcho** [comment]
I have same issue on CentOS 7:
[root@rocket opt]# yum info nodejs
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: mirrors.neterra.net
 * epel: mirrors.neterra.net
 * extras: mirrors.neterra.net
 * updates: mirrors.neterra.net
Installed Packages
Name        : nodejs
Arch        : x86_64
Epoch       : 2
Version     : 12.16.1
Release     : 1nodesource
Size        : 66 M
Repo        : installed
From repo   : nodesource
Summary     : JavaScript runtime
URL         : http://nodejs.org
License     : MIT and ASL 2.0 and ISC and BSD
Description : Node.js is a platform built on Chrome's JavaScript runtime
            : for easily building fast, scalable network applications.
            : Node.js uses an event-driven, non-blocking I/O model that
            : makes it lightweight and efficient, perfect for data-intensive
            : real-time applications that run across distributed devices.

My steps are:
Update nodejs from 8x to 12 from official nodejs repo for CentOS 7.
[root@rocket ~]# curl -L https://releases.rocket.chat/latest/download -o /tmp/rocket.chat.tgz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   223  100   223    0     0    273      0 --:--:-- --:--:-- --:--:--   273
100  146M  100  146M    0     0  43.5M      0  0:00:03  0:00:03 --:--:-- 74.9M
[root@rocket ~]# tar -xzf /tmp/rocket.chat.tgz -C /tmp
[root@rocket ~]# cd /tmp/bundle/programs/server && npm install
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142

> fibers@4.0.3 install /tmp/bundle/programs/server/node_modules/fibers
> node build.js || nodejs build.js

make: Entering directory `/tmp/bundle/programs/server/node_modules/fibers/build'
  CXX(target) Release/obj.target/fibers/src/fibers.o
  CXX(target) Release/obj.target/fibers/src/coroutine.o
  CC(target) Release/obj.target/fibers/src/libcoro/coro.o
  SOLINK_MODULE(target) Release/obj.target/fibers.node
  COPY Release/fibers.node
make: Leaving directory `/tmp/bundle/programs/server/node_modules/fibers/build'
Installed in `/tmp/bundle/programs/server/node_modules/fibers/bin/linux-x64-57-glibc/fibers.node`
npm WARN lifecycle meteor-dev-bundle@~install: cannot run in wd meteor-dev-bundle@ node npm-rebuild.js (wd=/tmp/bundle/programs/server)
added 147 packages from 122 contributors and audited 305 packages in 9.123s

1 package is looking for funding
  run `npm fund` for details

found 0 vulnerabilities

[root@rocket server]# sudo mv /tmp/bundle /opt/Rocket.Chat
[root@rocket server]# sudo chown -R rocketchat:rocketchat /opt/Rocket.Chat
[root@rocket server]# sudo systemctl start rocketchat

error: 
Feb 25 13:57:13 rocket systemd: Started The Rocket.Chat server.
Feb 25 13:57:13 rocket rocketchat: Meteor requires Node v12.0.0 or later.
Feb 25 13:57:13 rocket systemd: rocketchat.service: main process exited, code=exited, status=1/FAILURE
Feb 25 13:57:13 rocket systemd: Unit rocketchat.service entered failed state.
Feb 25 13:57:13 rocket systemd: rocketchat.service failed.

**Comment by sflyer** [comment]
ubuntu 18.04.
Follow instructions in https://rocket.chat/docs/installation/updating/
after sudo systemctl start rocketchat get error:

● rocketchat.service - Rocket.Chat server
   Loaded: loaded (/etc/systemd/system/rocketchat.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Thu 2020-03-05 13:46:13 +05; 8s ago
  Process: 10264 ExecStart=/usr/local/bin/node /opt/rocket/Rocket.Chat/main.js (code=exited, status=1/FAILURE)
 Main PID: 10264 (code=exited, status=1/FAILURE)

мар 05 13:46:13 chat systemd[1]: Started Rocket.Chat server.
мар 05 13:46:13 chat rocketchat[10264]: Meteor requires Node v12.0.0 or later.
мар 05 13:46:13 chat systemd[1]: rocketchat.service: Main process exited, code=exited, status=1/FAILURE
мар 05 13:46:13 chat systemd[1]: rocketchat.service: Failed with result 'exit-code'.

**Comment by jtorrex** [comment]
`● rocketchat.service - The Rocket.Chat server
   Loaded: loaded (/lib/systemd/system/rocketchat.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Fri 2020-03-06 01:33:42 CET; 1s ago
  Process: 16559 ExecStart=/usr/local/bin/node /opt/Rocket.Chat/main.js (code=exited, status=1/FAILURE)
 Main PID: 16559 (code=exited, status=1/FAILURE)

Mar 06 01:33:41 opsrocketchat01 systemd[1]: Started The Rocket.Chat server.
Mar 06 01:33:42 opsrocketchat01 rocketchat[16559]: Meteor requires Node v12.0.0 or later.
Mar 06 01:33:42 opsrocketchat01 systemd[1]: rocketchat.service: Main process exited, code=exited, status=1/FAILURE
Mar 06 01:33:42 opsrocketchat01 systemd[1]: rocketchat.service: Unit entered failed state.
Mar 06 01:33:42 opsrocketchat01 systemd[1]: rocketchat.service: Failed with result 'exit-code'.`

Same case here and updated Node from official repositories on Debian:

`# nodejs -v
v12.16.1`

**Comment by Trans-IP** [comment]
Same issue for me trying to upgrade to the latest release

# nodejs -v
v12.16.1

# npm -v
6.13.4

# cat /etc/issue
Debian GNU/Linux 9 \n \l

**Comment by Trans-IP** [comment]
Hey I think I figured it out!
From the debian install guide:

`sudo npm install -g inherits n && sudo n 12.14.0`

This seems to do the trick with the latest release (3.03 in my case)

**Comment by jtorrex** [comment]
> I have same issue on CentOS 7:
> [root@rocket opt]# yum info nodejs
> Loaded plugins: fastestmirror
> Loading mirror speeds from cached hostfile
> 
>     * base: mirrors.neterra.net
> 
>     * epel: mirrors.neterra.net
> 
>     * extras: mirrors.neterra.net
> 
>     * updates: mirrors.neterra.net
>       Installed Packages
>       Name        : nodejs
>       Arch        : x86_64
>       Epoch       : 2
>       Version     : 12.16.1
>       Release     : 1nodesource
>       Size        : 66 M
>       Repo        : installed
>       From repo   : nodesource
>       Summary     : JavaScript runtime
>       URL         : http://nodejs.org
>       License     : MIT and ASL 2.0 and ISC and BSD
>       Description : Node.js is a platform built on Chrome's JavaScript runtime
>       : for easily building fast, scalable network applications.
>       : Node.js uses an event-driven, non-blocking I/O model that
>       : makes it lightweight and efficient, perfect for data-intensive
>       : real-time applications that run across distributed devices.
> 
> 
> My steps are:
> Update nodejs from 8x to 12 from official nodejs repo for CentOS 7.
> Try update to latest rocket.chat version:
> [root@rocket ~]# curl -L https://releases.rocket.chat/latest/download -o /tmp/rocket.chat.tgz
> % Total % Received % Xferd Average Speed Time Time Time Current
> Dload Upload Total Spent Left Speed
> 100 223 100 223 0 0 273 0 --:--:-- --:--:-- --:--:-- 273
> 100 146M 100 146M 0 0 43.5M 0 0:00:03 0:00:03 --:--:-- 74.9M
> [root@rocket ~]# tar -xzf /tmp/rocket.chat.tgz -C /tmp
> [root@rocket ~]# cd /tmp/bundle/programs/server && npm install
> npm WARN deprecated request@2.88.2: request has been deprecated, see [request/request#3142](https://github.com/request/request/issues/3142)
> 
> > fibers@4.0.3 install /tmp/bundle/programs/server/node_modules/fibers
> > node build.js || nodejs build.js
> 
> make: Entering directory `/tmp/bundle/programs/server/node_modules/fibers/build' CXX(target) Release/obj.target/fibers/src/fibers.o CXX(target) Release/obj.target/fibers/src/coroutine.o CC(target) Release/obj.target/fibers/src/libcoro/coro.o SOLINK_MODULE(target) Release/obj.target/fibers.node COPY Release/fibers.node make: Leaving directory `/tmp/bundle/programs/server/node_modules/fibers/build'
> Installed in `/tmp/bundle/programs/server/node_modules/fibers/bin/linux-x64-57-glibc/fibers.node`
> npm WARN lifecycle meteor-dev-bundle@~install: cannot run in wd meteor-dev-bundle@ node npm-rebuild.js (wd=/tmp/bundle/programs/server)
> added 147 packages from 122 contributors and audited 305 packages in 9.123s
> 
> 1 package is looking for funding
> run `npm fund` for details
> 
> found 0 vulnerabilities
> 
> [root@rocket server]# sudo mv /tmp/bundle /opt/Rocket.Chat
> [root@rocket server]# sudo chown -R rocketchat:rocketchat /opt/Rocket.Chat
> [root@rocket server]# sudo systemctl start rocketchat
> 
> error:
> Feb 25 13:57:13 rocket systemd: Started The Rocket.Chat server.
> Feb 25 13:57:13 rocket rocketchat: Meteor requires Node v12.0.0 or later.
> Feb 25 13:57:13 rocket systemd: rocketchat.service: main process exited, code=exited, status=1/FAILURE
> Feb 25 13:57:13 rocket systemd: Unit rocketchat.service entered failed state.
> Feb 25 13:57:13 rocket systemd: rocketchat.service failed.



> ubuntu 18.04.
> Follow instructions in https://rocket.chat/docs/installation/updating/
> after sudo systemctl start rocketchat get error:
> 
> ● rocketchat.service - Rocket.Chat server
> Loaded: loaded (/etc/systemd/system/rocketchat.service; enabled; vendor preset: enabled)
> Active: failed (Result: exit-code) since Thu 2020-03-05 13:46:13 +05; 8s ago
> Process: 10264 ExecStart=/usr/local/bin/node /opt/rocket/Rocket.Chat/main.js (code=exited, status=1/FAILURE)
> Main PID: 10264 (code=exited, status=1/FAILURE)
> 
> мар 05 13:46:13 chat systemd[1]: Started Rocket.Chat server.
> мар 05 13:46:13 chat rocketchat[10264]: Meteor requires Node v12.0.0 or later.
> мар 05 13:46:13 chat systemd[1]: rocketchat.service: Main process exited, code=exited, status=1/FAILURE
> мар 05 13:46:13 chat systemd[1]: rocketchat.service: Failed with result 'exit-code'.

Hi all,

In my case. I discovered that Node was successfully upgraded on my Debian box. You also can test it looking for the paths where the bins reside:

```# whereis node
node: /usr/bin/node /usr/local/bin/node /usr/include/node /usr/share/man/man1/node.1.gz```

or

```# whereis nodejs
nodejs: /usr/bin/nodejs /usr/share/man/man1/nodejs.1.gz```

Bins under /usr/bin returns that are correctly updated, and this version it's what Rocket expects:

```
#/usr/bin/node -v
v12.16.1 
```
or

```#/usr/bin/nodejs -v
v12.16.1
```
My problem is that on the env of the user that I'm using, on the var $PATH, /usr/local/bin it's checked before /usr/bin:

```PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin```

If I check the version that reside on /usr/local/bin, the result is:

```
#/usr/local/bin/node -v
v8.X.X
```
Seems that the node upgrade doesn't link or change the bin that reside under /usr/local/bin and this is the first on been checked when we try to start the service, so this impedes that RocketChat service starts normally.

In my case I solved this issue removing the bin for the old version and pointing a link to the upgraded version:

```rm /usr/local/bin/node && ln -s /usr/bin/node /usr/local/bin/node
```
After this you should obtain:

```
# /usr/local/bin/node -v
v12.16.1
```

And the RocketChat starts normally.

Please, check your env and bring me your feedback.

Regards!

**Comment by sflyer** [comment]
> Hey I think I figured it out!
> From the debian install guide:
> 
> `sudo npm install -g inherits n && sudo n 12.14.0`
> 
> This seems to do the trick with the latest release (3.03 in my case)

Work for me, thanks!

**Comment by wkriesch** [comment]
Hi! same problem here, but i tried, npm version, old versions and newest version, nothing, i've tried update working version!, nothing.. i've tried install with, CentOS, Debian, Ubuntu.. and nothing

all that! same error...

[root@localhost server]# cd /tmp/bundle/programs/server && npm install
npm WARN lifecycle meteor-dev-bundle@~install: cannot run in wd meteor-dev-bundle@ node npm-rebuild.js (wd=/tmp/bundle/programs/server)
audited 305 packages in 3.388s

1 package is looking for funding
  run `npm fund` for details

found 1 moderate severity vulnerability
  run `npm audit fix` to fix them, or `npm audit` for details



Plz Help US!

**Comment by miahalps** [comment]
Hi, I am having this same problem and tried the above--still stuck! Any insight?

**Comment by geekgonecrazy** [comment]
It looks like we need to handle the node version using something other than the package manager :(

**Comment by amoebav** [comment]
> It looks like we need to handle the node version using something other than the package manager :(
My problem was that I was using yum in Centos to update node. Nope, I needed to use npm which is what I originally used to install nodejs. Centos repository doesn't have current stable version of node.

I was able to upgrade from node v8.16.0 to the current stable release v12.18.3 with no problems.

I ran these commands to update:
npm cache clean -f
npm install -g n
n stable
